src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -635,6 +635,7 @@ private static ushort RoundPackToHalf(bool sign, short exp, ushort sig)
                 {
                     sig = (ushort)ShiftRightJam(sig, -exp);
                     exp = 0;
+                    roundBits = sig & 0xF;
```

Comment:
The other issue was that this was missing adjusting the roundBits based on the updated sig.

Reply:
The original implementation was doing the correct thing: https://github.com/ucb-bar/berkeley-softfloat-3/blob/b64af41c3276f97f0e181920400ee056b9c88037/source/s_roundPackToF16.c#L77

I imagine this was accidentally dropped when eliding the unnecessary exception handling logic.